File: android/src/main/kotlin/com/example/mango_player/MangoPlayerPlugin.kt
package com.example.mango_player

import androidx.annotation.NonNull
import io.flutter.embedding.engine.plugins.FlutterPlugin
import com.example.mango_player.platform.MethodChannelHandler
import com.example.mango_player.platform.EventChannelHandler
import com.example.mango_player.platform.TextureRegistryHandler
import com.example.mango_player.ijkplayer.IJKPlayerManager

class MangoPlayerPlugin: FlutterPlugin {
  private lateinit var methodChannelHandler: MethodChannelHandler
  private lateinit var eventChannelHandler: EventChannelHandler
  private lateinit var textureRegistryHandler: TextureRegistryHandler
  private lateinit var playerManager: IJKPlayerManager

  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    playerManager = IJKPlayerManager(flutterPluginBinding.applicationContext)
    
    textureRegistryHandler = TextureRegistryHandler(flutterPluginBinding.textureRegistry)
    playerManager.setTextureRegistryHandler(textureRegistryHandler)
    
    methodChannelHandler = MethodChannelHandler(flutterPluginBinding.binaryMessenger, playerManager, textureRegistryHandler)
    eventChannelHandler = EventChannelHandler(flutterPluginBinding.binaryMessenger, playerManager)
    
    methodChannelHandler.startListening()
    eventChannelHandler.startListening()
    textureRegistryHandler.startListening(flutterPluginBinding.binaryMessenger)
  }

  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
    methodChannelHandler.stopListening()
    eventChannelHandler.stopListening()
    textureRegistryHandler.stopListening()
    playerManager.release()
  }
}

File: ios/Classes/MangoPlayerPlugin.swift
import Flutter
import UIKit

public class MangoPlayerPlugin: NSObject, FlutterPlugin {
  var methodChannelHandler: MethodChannelHandler?
  var eventChannelHandler: EventChannelHandler?
  var textureRegistryHandler: TextureRegistryHandler?
  var playerManager: IJKPlayerManager?

  public static func register(with registrar: FlutterPluginRegistrar) {
    let instance = MangoPlayerPlugin()
    instance.setup(registrar: registrar)
  }
  
  private func setup(registrar: FlutterPluginRegistrar) {
    playerManager = IJKPlayerManager()
    
    textureRegistryHandler = TextureRegistryHandler(registry: registrar.textures())
    playerManager?.setTextureRegistryHandler(handler: textureRegistryHandler)
    
    methodChannelHandler = MethodChannelHandler(messenger: registrar.messenger(), playerManager: playerManager!, textureRegistryHandler: textureRegistryHandler!)
    eventChannelHandler = EventChannelHandler(messenger: registrar.messenger(), playerManager: playerManager!)
    
    methodChannelHandler?.startListening()
    eventChannelHandler?.startListening()
    textureRegistryHandler?.startListening(messenger: registrar.messenger())
  }
  
  public func detachFromEngine(for registrar: FlutterPluginRegistrar) {
    methodChannelHandler?.stopListening()
    eventChannelHandler?.stopListening()
    textureRegistryHandler?.stopListening()
    playerManager?.release()
  }
}
